### PR TITLE
multi: consolidate trivial PRs that are stuck on CI pipeline steps

### PR DESCRIPTION
**Pull request by guggero**
Replaces https://github.com/lightningnetwork/lnd/pull/5578.
Replaces https://github.com/lightningnetwork/lnd/pull/5653.
Replaces https://github.com/lightningnetwork/lnd/pull/5942.
Replaces https://github.com/lightningnetwork/lnd/pull/5947.
Replaces https://github.com/lightningnetwork/lnd/pull/6051.
Replaces https://github.com/lightningnetwork/lnd/pull/5747.
Replaces https://github.com/lightningnetwork/lnd/pull/5848.
Replaces https://github.com/lightningnetwork/lnd/pull/2621.
Replaces https://github.com/lightningnetwork/lnd/pull/5094.
Replaces https://github.com/lightningnetwork/lnd/pull/5606.

### CODE REVIEW

**Review by Roasbeef — APPROVED** [review]
LGTM 🦚

**Comment by Roasbeef** [comment]
Has a conflict.

**Comment by guggero** [comment]
Rebased.

File: config.go [hunk 1]
```diff
@@ -633,11 +633,22 @@ func LoadConfig(interceptor signal.Interceptor) (*Config, error) {
...
+	case configFileDir != DefaultLndDir &&
+		configFilePath == DefaultConfigFile:
+
+		configFilePath = filepath.Join(
+			configFileDir, lncfg.DefaultConfigFilename,
+		)
+
+	// User did specify an explicit --configfile, so we check that it does
+	// exist under that path to avoid surprises.
+	case configFilePath != DefaultConfigFile:
```
Comment (carlaKC) [inline config.go:648]: nice!

File: channeldb/payments.go [hunk 2]
```diff
@@ -731,7 +731,7 @@ func fetchPaymentWithSequenceNumber(tx kvdb.RTx, paymentHash lntypes.Hash,
 // DeletePayment deletes a payment from the DB given its payment hash. If
 // failedHtlcsOnly is set, only failed HTLC attempts of the payment will be
 // deleted.
-func (d *DB) DeletePayment(paymentHash lntypes.Hash, failedHtlcsOnly bool) error { // nolint:interfacer
+func (d *DB) DeletePayment(paymentHash lntypes.Hash, failedHtlcsOnly bool) error {
```
Comment (carlaKC) [inline channeldb/payments.go:734]: Wrap at 80 while we're here?

**Review by carlaKC — COMMENTED** [review]
Config file change has borked itests: `failed to load config: specified config file does not exist in /tmp/lndtest-node679047399/data`

**Comment by guggero** [comment]
Fixed the itests, seems that `--configfile` flag in the tests was incorrect the whole time!

**Review by carlaKC — APPROVED** [review]
Assuming that other itest failures are unrelated!

**Pull request merged**